### ISSUE TITLE
Why is forward method for ANP maximizing LogLikelihood for Context labels?

### ISSUE THREAD

**Issue by yair-schiff** [juho-lee/bnp#1]
In the [ANP](https://github.com/juho-lee/bnp/blob/master/regression/models/anp.py) implementation, during training mode, the full set (i.e. context + target) is being passed to `predict` meaning that log likelihood is being maximized for both target and context labels.
```python
        if self.training:
            pz = self.lenc(batch.xc, batch.yc)
            qz = self.lenc(batch.x, batch.y)
            z = qz.rsample() if num_samples is None else \
                    qz.rsample([num_samples])
            py = self.predict(batch.xc, batch.yc, batch.x,
                    z=z, num_samples=num_samples)


            if num_samples > 1:
                # K * B * N
                recon = py.log_prob(stack(batch.y, num_samples)).sum(-1)
                # K * B
                log_qz = qz.log_prob(z).sum(-1)
                log_pz = pz.log_prob(z).sum(-1)


                # K * B
                log_w = recon.sum(-1) + log_pz - log_qz


                outs.loss = -logmeanexp(log_w).mean() / batch.x.shape[-2]
```

Shouldn't this [line](https://github.com/juho-lee/bnp/blob/4cf4b92b38b3ac2aa4bfbbd81de596c9bf7e9645/regression/models/anp.py#L65):
```python
            py = self.predict(batch.xc, batch.yc, batch.x,
```
be changed to this
```python
            py = self.predict(batch.xc, batch.yc, batch.xt,
```
i.e. `batch.x` --> `batch.xt` in 3rd argument of `predict`?
